rest_lights.cpp

```diff
@@ -1973,101 +1973,113 @@ int DeRestPluginPrivate::setWindowCoveringState(const ApiRequest &req, ApiRespon
+            rsp.list.append(rspItem);
+        }
         else
         {
-            rsp.list.append(errorToMap(ERR_PARAMETER_NOT_AVAILABLE, QString("/lights/%1/state/on").arg(id), QString("parameter, not available")));
-            rsp.httpStatus = HttpStatusBadRequest;
-            return REQ_READY_SEND;
+            rsp.list.append(errorToMap(ERR_INTERNAL_ERROR, QString("/lights/%1").arg(id), QString("Internal error, %1").arg(ERR_BRIDGE_BUSY)));
         }
     }
-    else if (map.contains("alert"))
+
+    if (hasOn)
     {
-        if (map["alert"].type() == QVariant::String)
+        qint8 button = DP_IDENTIFIER_BUTTON_1;
+        QByteArray data;
+
+        //Retreive Fake endpoint, and change button value
+        uint8_t ep = taskRef.lightNode->haEndpoint().endpoint();
+        if (ep == 0x02) { button = DP_IDENTIFIER_BUTTON_2; }
+        if (ep == 0x03) { button = DP_IDENTIFIER_BUTTON_3; }
+
+        //Use only the first endpoint for command
+        taskRef.req.setDstEndpoint(0x01);
+
+        DBG_Printf(DBG_INFO, "Tuya debug 10: EP: %d ID : %s\n", ep, qPrintable(id));
+
+        if (targetOn)
+        {
+            data = QByteArray("\x01",1);
+        }
+        else
         {
-            QByteArray data("\x00", 1);
+            data = QByteArray("\x00",1);
+        }
 
-            if (map["alert"].toString() == "lselect")
-            {
-                data = QByteArray("\x01",1);
-            }
+        ok = sendTuyaRequest(taskRef, TaskTuyaRequest, DP_TYPE_BOOL, button, data);
 
-            if (sendTuyaRequest(taskRef, TaskTuyaRequest, DP_TYPE_BOOL, DP_IDENTIFIER_ALARM, data))
-            {
-            }
-            else
-            {
-                rsp.list.append(errorToMap(ERR_INTERNAL_ERROR, QString("/lights/%1").arg(id), QString("Internal error, %1").arg(ERR_BRIDGE_BUSY)));
-            }
+        if (ok)
+        {
+            QVariantMap rspItem;
+            QVariantMap rspItemState;
+            rspItemState[QString("/lights/%1/state/on").arg(id)] = targetOn;
+            rspItem["success"] = rspItemState;
+            rsp.list.append(rspItem);
         }
         else
         {
-            rsp.list.append(errorToMap(ERR_PARAMETER_NOT_AVAILABLE, QString("/lights/%1/state/alert").arg(id), QString("parameter, not available")));
-            rsp.httpStatus = HttpStatusBadRequest;
-            return REQ_READY_SEND;
+            rsp.list.append(errorToMap(ERR_INTERNAL_ERROR, QString("/lights/%1").arg(id), QString("Internal error, %1").arg(ERR_BRIDGE_BUSY)));
         }
+
     }
-    else
+    
+    if (!hasOn && !hasBri)
     {
-        rsp.list.append(errorToMap(ERR_PARAMETER_NOT_AVAILABLE, QString("/lights/%1/state/on").arg(id), QString("parameter not available")));
+        rsp.list.append(errorToMap(ERR_PARAMETER_NOT_AVAILABLE, QString("/lights/%1/state").arg(id), QString("parameter not available")));
         rsp.httpStatus = HttpStatusBadRequest;
         return REQ_READY_SEND;
     }

```

Comment:
Which parameter is not available? :)
I think each not handled parameter should be added as error entry in `rsp.list`.

Reply:
A little better